dora/core/server/worker/src/main/java/alluxio/worker/task/CopyHandler.java

```diff
@@ -81,10 +81,13 @@ public static void copy(Route route, WriteOptions writeOptions,
     } catch (Exception e) {
       throw AlluxioRuntimeException.from(e);
     }
-    if (dstStatus != null && !writeOptions.getOverwrite()) {
+    if (dstStatus != null && dstStatus.isFolder() && sourceStatus.isFolder()) {
```

Comment:
does this do the expected behavior if:
- src is a directory with files a, b, and c
- dst is a directory with equivalent files a and b, but does not contain c
- run distCp from src to dst

with this new condition, i would think file c does not get copied to dst? but does it make sense to do a recursive check if there's no recursive flag? i'm not sure what the expected behavior should be @jja725

Reply:
file c would copied to dst. This is single inode operation. So it would only skip copy directory inode.

Reply:
sounds good, thanks for clarifying